component/component/statefuleset-resize-controller.jsonnet

```diff
@@ -66,10 +66,47 @@ local deployment = loadManifest('config/manager/manager.yaml') + {
   },
 };
 
+local resizeServiceAccount = kube.ServiceAccount('sa-sts-deleter') + {
+  metadata+: {
+    namespace: params.services.controlNamespace,
+  },
+};
+
+local resizeClusterRole = kube.ClusterRole('appcat:job:resizejob') {
+  rules: [
+    {
+      apiGroups: [ 'helm.crossplane.io' ],
+      resources: [ 'releases' ],
+      verbs: [ 'get', 'list', 'watch', 'update', 'patch', 'create', 'delete' ],
+    },
+    {
+      apiGroups: [ 'apps' ],
+      resources: [ 'statefulsets' ],
+      verbs: [ 'delete', 'get', 'watch', 'list', 'update', 'patch' ],
+    },
+    {
+      apiGroups: [ 'helm.crossplane.io' ],
+      resources: [ 'releases' ],
+      verbs: [ 'update', 'get' ],
+    },
+    {
+      apiGroups: [ '' ],
+      resources: [ 'pods' ],
+      verbs: [ 'list', 'get', 'update', 'delete' ],
+    },
+  ],
+};
+
+local resizeClusterRoleBinding = kube.ClusterRoleBinding('appcat:job:resizejob') + {
+  roleRef_: resizeClusterRole,
+  subjects_: [ resizeServiceAccount ],
+};
+
 // Curently we only need this for redis.
-if params.services.vshn.enabled && params.services.vshn.redis.enabled then {
+if params.services.vshn.enabled && params.services.vshn.stsResizer.enabled then {
```

Comment:
Why did you change this? The stsResizer is not really usable on its own, it should be automatically installed when services are enabled that depend on it, otherwise it's yet another config flag we actively need to think about. It's already complex enough to set all the right flags to enable a service, let's not increase the configuration complexity if not necessary.

Reply:
Switched it back to the original.
However, this appproach is also not verify feasable in the future, as we would have to include every future helm service in this if clause in the future.